### PR TITLE
Add request headers to saga transaction, add mail service

### PR DESCRIPTION
**Pull request by igordumchykov**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/MailService.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public class MailService {
+
+    @Autowired
+    private final MailProviderService mailProviderService;
+
+    public void sendEmail(String to, String subject, String content, String from) {
+        log.debug("Send email[multipart '{}' and html '{}'] to '{}' with subject '{}' and content={}",
+            false, true, to, subject, content);
+
+        JavaMailSender javaMailSender = mailProviderService.getJavaMailSender(TenantKey.SUPER_TENANT_KEY_VALUE.toUpperCase());
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/MailService.java]: get tenant from TenantContextHolder

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/ServiceApiImpl.java [hunk 2]
```diff
@@ -32,10 +37,24 @@
         if (isNotEmpty(service.getServiceCharacteristic())) {
             service.getServiceCharacteristic().forEach(ch -> params.put(ch.getName(), ch.getValue()));
         }
+
+        //if service processes LEP`s asynchronously, RequestContextHolder will not be able to get request
+        //attributes from thread that will execute LEP
+        ServletRequestAttributes requestAttributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/ServiceApiImpl.java]: maybe better to extract function like `addAllHeaders` ?

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/MailService.java (hunk 1 above)
Comment (igordumchykov) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/MailService.java]: done

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/ServiceApiImpl.java (hunk 2 above)
Comment (igordumchykov) [reply src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/ServiceApiImpl.java]: agree, done

**Pull request merged**